content/en/blog/community/spinkube-kind-rd/index.md

```diff
@@ -0,0 +1,328 @@
+  default_runtime_name = "runc"
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+  # set default runtime handler to v2, which has a per-pod shim
+  runtime_type = "io.containerd.runc.v2"
+  # Generated by "ctr oci spec" and modified at base container to mount poduct_uuid
+  base_runtime_spec = "/etc/containerd/cri-base.json"
+  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+    # use systemd cgroup by default
+    SystemdCgroup = true
+
+# Setup a runtime with the magic name ("test-handler") used for Kubernetes
+# runtime class tests ...
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.test-handler]
+  # same settings as runc
+  runtime_type = "io.containerd.runc.v2"
+  base_runtime_spec = "/etc/containerd/cri-base.json"
+  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.test-handler.options]
+    SystemdCgroup = true
+
+[plugins."io.containerd.grpc.v1.cri"]
+  # use fixed sandbox image
+  sandbox_image = "registry.k8s.io/pause:3.7"
+  # allow hugepages controller to be missing
+  # see https://github.com/containerd/cri/pull/1501
+  tolerate_missing_hugepages_controller = true
+  # restrict_oom_score_adj needs to be true when running inside UserNS (rootless)
+  restrict_oom_score_adj = false
+  
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.spin]
+  runtime_type = "/usr/local/bin/containerd-shim-spin-v2"
+
+```
+
+> NOTE: this file is a copy of the original one that can be found inside the KinD container. The only addition to the file is the declaration of the `spin` plugin (the last 2 lines)
+
+### KinD config file
+
+The second configuration file is related to KinD and will be used when creating a new cluster:
+
+- Create a file in your `$HOME` directory called `kind-spin.yaml` (for example)
+  - You can create it inside a directory, however it still should be located in your `$HOME` directory
+  - The location will be important when creating the KinD cluster
+
+**Windows Users ONLY**
+
+```yaml
+@"
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+  - hostPath: /root/kind-spin/config.toml
+    containerPath: /etc/containerd/config.toml
+  - hostPath: /usr/local/containerd-shims/containerd-shim-spin-v2
+    containerPath: /usr/local/bin/containerd-shim-spin-v2
+"@ | rdctl shell -- tee -a /root/test.yaml
+```
+
+**Linux and MacOS Users ONLY**
+
+```yaml
+cat <<EOF > $HOME/kind-spin.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+  - hostPath: /home/${USER}/config.toml
+    containerPath: /etc/containerd/config.toml
+  - hostPath: /usr/local/containerd-shims/containerd-shim-spin-v2
+    containerPath: /usr/local/bin/containerd-shim-spin-v2
+EOF
+```
+
+#### Connecting the dots (2)
+
+Rancher Desktop leverages 2 different technologies depending on the OS its installed.
+
+On Windows, WSL2 will be used and for Linux and MacOS, Lima is the preferred choice.
```

Comment:
This is where I might have gotten a little lost.  I'm on MacOS and was unclear on where I should be creating the KinD cluster below.  I had Rancher Desktop running as instructed above, but no Lima VM since I didn't see any specific steps.  Thus, I attempted to create the kind cluster on my host machine which failed as I don't have `nerdctl` installed locally (indeed, there doesn't appear to be a [local MacOS option]( https://github.com/containerd/nerdctl?tab=readme-ov-file#macos)) and this is what I encountered:

```console
$ kind create cluster --config=./kind-spin.yaml
using nerdctl due to KIND_EXPERIMENTAL_PROVIDER
ERROR: failed to create cluster: failed to get nerdctl info: command "nerdctl info --format '{{json .}}'" failed with error: exec: "nerdctl": executable file not found in $PATH
Command Output:
```

Did I maybe miss a step?  It seems like I should be in a Lima VM ... or the Rancher Desktop VM?

Reply:
First, thanks a lot for the great feedback, I'll correct them.

As for the LIMA based install, you won't see it by default indeed. Rancher Desktop doesn't expose the `lima` binary, and the LIMA VM is not created in the default place (see https://github.com/lima-vm/lima/discussions/1517 for details).

I guess there's the step in prerequisites about choosing `containerd engine` that was missed. I truly it's not your fault and I suppose it doesn't stand out enough.

Before I change it, can I kindly ask you to check if you choose `containerd` and enabled `wasm` in the Rancher Desktop preferences?

Reply:
@nunix thanks for checking, this led me to the issue.  Although I did have `containerd` selected and `wasm` enabled, I noticed there were some alerts in the "Diagnostics" tab, including the following:
```
The ~/.rd/bin directory has not been added to the PATH, so command-line utilities are not configured in your bash shell.
```
Sure enough, once I updated my PATH, `nerdctl` was available to use!  Nice.

Not sure if this is worth a special mention or not -- could definitely chalk it up to my being a novice Rancher Desktop user :)